subsystems/shootingsubsystem.py

```diff
@@ -0,0 +1,171 @@
+            constants.kShootingMotorId,
+            constants.kSimShootingMotorPort,
+            constants.kShootingPGain,
+            constants.kShootingIGain,
+            constants.kShootingDGain,
+            constants.kShootingPIDSlot,
+            constants.kShootingMotorInverted,
+        )
+        self.hoodMotor = Falcon(
+            constants.kHoodMotorName,
+            constants.kHoodMotorId,
+            constants.kSimHoodMotorPort,
+            constants.kHoodPGain,
+            constants.kHoodIGain,
+            constants.kHoodDGain,
+            constants.kHoodPIDSlot,
+            constants.kHoodMotorInverted,
+        )
+
+        self.initializationMinimum = 0
+        self.initializationMaximum = 0
+
+        self.targetTurretAngle = Rotation2d()
+        self.targetHoodAngle = Rotation2d()
+        self.targetWheelSpeed = 0
+
+    def setAsStartingPosition(self) -> None:
+        self.hoodMotor.setCurrentEncoderPulseCount(constants.kHoodStartingAngle)
+        if (
+            RobotBase.isReal()
+        ):  # only possible to calibrate the real robot, sim is perfect by default
+            turretRealPosition = map_range(
+                self.turretMotor.getPosition(),
+                self.initializationMinimum,
+                self.initializationMaximum,
+                constants.kTurretMinimumAngle.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+                constants.kTurretMaximumAngle.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+            )
+            self.turretMotor.setCurrentEncoderPulseCount(turretRealPosition)
+            self.turretMotor.setBrakeMode()
+
+    def periodic(self) -> None:
+        self.initializationMinimum = min(
+            self.initializationMinimum, self.turretMotor.getPosition()
+        )
+        self.initializationMaximum = max(
+            self.initializationMaximum, self.turretMotor.getPosition()
+        )
+        if not SmartDashboard.getBoolean(constants.kShootingManualModeKey, False):
+            SmartDashboard.putNumber(
+                constants.kShootingWheelSpeedKey, self.getWheelSpeed()
+            )
+            SmartDashboard.putNumber(
+                constants.kShootingHoodAngleKey, self.getHoodAngle().degrees()
+            )
+            SmartDashboard.putNumber(
+                constants.kShootingTurretAngleKey, self.getTurretRotation().degrees()
+            )
+        SmartDashboard.putBoolean(constants.kShootingOnTargetKey, self.onTarget())
+
+    def onTarget(self) -> bool:
+        if (
+            abs(self.getWheelSpeed() - self.targetWheelSpeed)
+            <= constants.kWheelSpeedTolerence
+            and abs((self.getHoodAngle() - self.targetHoodAngle).radians())
+            <= constants.kHoodAngleTolerence.radians()
+            and abs((self.getTurretRotation() - self.targetTurretAngle).radians())
+            <= constants.kTurretAngleTolerence.radians()
```

Comment:
Let's calculate each onTarget boolean separately (turretOnTarget, hoodOnTarget, flywheelOnTarget) and output them to SD for easier debugging.